### DISCUSSION TITLE
Problem med nedladdning av film från SVT Play

### DISCUSSION THREAD

**Discussion by GigaCat53** [stefansundin/privatkopiera#135] [Q&A] [answered]
Testade nyligen att ladda ned en film från SVT play. Efter att ca en minut har gått så spottar terminalen ut ett felmeddelande och avslutar processen. 
https://www.svtplay.se/video/37206765/21-jump-street?id=jXkvvr5

![image](https://user-images.githubusercontent.com/117866961/200951192-d11fe59b-ba58-459d-9907-abb4f5656e15.png)

Bilden i textformat
`[hls @ 00000297f0168040] Opening 'crypto+https://svt-vod-secure-1c.akamaized.net/d0/secure/20221027/e6b27d00-6d74-4c8d-abde-db848b2b824e/hls-audio-sv-aac-2ch-192/hls-audio-sv-aac-2ch-192-23.aac?aka_me_session_id=AAAAAAAAAAAla21jAAAAAHxUrPb5blZMPuAkGZvpgE3hKTEx3rglFT+IhK6DtahQE80IrOCnPWC2u8rg1ZSaPKAEHm2K3pZv&aka_msn=22&aka_hls_version=4&aka_media_format_type=hls' for reading
[aac_adtstoasc @ 00000297eff22e40] Multiple RDBs per frame with CRC is not implemented. Update your FFmpeg version to the newest one from Git. If the problem still occurs, it means that your file has a feature which has not been implemented.
[matroska @ 00000297f01b9b40] Error applying bitstream filters to an output packet for stream #1: Not yet implemented in FFmpeg, patches welcome
av_interleaved_write_frame(): Not yet implemented in FFmpeg, patches welcome
Error muxing a packet for output file #0
frame= 2117 fps=216 q=-1.0 Lsize=   30391kB time=00:01:24.72 bitrate=2938.6kbits/s speed=8.63x
video:28376kB audio:2001kB subtitle:0kB other streams:0kB global headers:0kB muxing overhead: 0.042232%
Conversion failed!`

Filen som blir kvar är en video på 1 minut och 24 sekunder med ljud och bild. Jag har uppdaterat FFmpeg till den sensate versionen, med det hjälpte inte. Testade även att ladda ned ett avsnitt på Rapport på 32 minuter. Fungerade felfritt. 

Någon som har någon aning om vart problemet kan ligga?
Har svt börjat med kryptering/DRM på Filmer?

**Comment by DuckRandom** [comment]
~~Det verkar uppstå med program som varnar att "Detta program är olämpligt för barn".~~

 Edit3: ^^Där hade jag fel, glöm det :)

Min lösning är att använda **yt-dlp**. svtplay-dl borde också fungera.
 
Edit:  Det kan vara ett gammalt problem som kommit tillbaka som har att göra med **aka_me_session_id**.
Jag fick det att fungera med en gammal ffmpeg som jag hade döpt till 'ffmpeg 4.2.2-funkar med mysko svtplay-grejen (aka_me_session_id)'.
Vad den gör annorlunda än nyare versioner har jag ingen aning om, tyvärr.

Edit2: Jag var tydligen trött i går. 
Från readme-filen : "Zeranoe FFmpeg Builds <http://ffmpeg.zeranoe.com/builds/> Build: ffmpeg-4.2.2-win64-static".
Edit4: Uppladdad här https://easyupload.io/0hx4bw i 30 dagar.

**Comment by Ordval** [comment]
[svtplay-dl ](https://svtplay-dl.se/archive/) är absolut en möjlighet, 
nu använder jag bara den (och Privatkopiera) för att få 
tag i MRL:erna `svtplay-dl --get-only-episode-url`  
(så vad gäller resten vet jag inte).

Din "ffmpeg-4.2.2-win64-static"-länk fungerar i alla fall 
inte för mig (men vi kan förstås leta själva).

**Reply by DuckRandom** [reply]
Jag ber om ursäkt, jag hade helt glömt denna tråd. Jag slängde upp filen här https://easyupload.io/0hx4bw 
Ligger bara kvar i 30 dagar.

**Comment by stefansundin** [comment, marked as answer]
EDIT: Jag gjorde en copy-paste. Det här är nog inte fixat. Eftersom det är en amerikansk film så antar jag att det är DRM-skyddat eller något.

**Reply by Ordval** [reply]
MRL:erna som jag skriver om ovan är huvudsakligen 
från SVT Nyheter. 
Med svtplay-dl kan en länk se ut så här: 
https://svt-vod-10f.akamaized.net/d0/world/20221120/190a77c6-e894-44ad-a285-a914eee844a2/dash-avc.mpd
(jag har brukat ändra ändelsen till: `hls-ts-avc.m3u8`)
`dash-avc.mpd` varieras ibland med `dash-hbbtv-avc.mpd` 
och är då den bättre, båda går att spela som de är.  

Jag letade efter någon utländsk film: "[Escape from Pretoria](https://www.svtplay.se/video/K5RoWXE/escape-from-pretoria)"
(tillg. till 25 dec., värst vad krångligt det var) och kan se 
att `dash-hbbtv-avc.mpd` även är tillämpbar här,
däremot har jag inte bemödat mig om att ladda ner.

**Comment by Ordval** [comment]
Så,  jag hade inte läst Stefan Sundins senaste 
kommentar på [SVTPlay problem #133](https://github.com/stefansundin/privatkopiera/discussions/135).
